File: android/src/main/java/org/ligi/passandroid/AppModule.java
package org.ligi.passandroid;

import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import com.squareup.moshi.Moshi;
import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;
import org.greenrobot.eventbus.EventBus;
import org.ligi.passandroid.json_adapter.ColorAdapter;
import org.ligi.passandroid.json_adapter.ZonedTimeAdapter;
import org.ligi.passandroid.model.AndroidFileSystemPassStore;
import org.ligi.passandroid.model.AndroidSettings;
import org.ligi.passandroid.model.PassStore;
import org.ligi.passandroid.model.Settings;

@Module
public class AppModule {

    private final App app;

    public AppModule(App app) {
        this.app = app;
    }

    @Singleton
    @Provides
    PassStore providePassStore(Settings settings, Moshi moshi,EventBus bus) {
        return new AndroidFileSystemPassStore(app, settings, moshi,bus);
    }

    @Singleton
    @Provides
    Moshi provideMoshi() {
        return new Moshi.Builder()
                .add(new ZonedTimeAdapter())
                .add(new ColorAdapter())
                .build();
    }

    @Singleton
    @Provides
    Settings provideSettings() {
        return new AndroidSettings(app);
    }

    @Singleton
    @Provides
    SharedPreferences provideSharedPreferences() {
        return PreferenceManager.getDefaultSharedPreferences(app);
    }

    @Singleton
    @Provides
    EventBus provideBus() {
        return EventBus.getDefault();
    }
}

File: android/src/main/java/org/ligi/passandroid/AppModule.kt
package org.ligi.passandroid

import android.content.SharedPreferences
import android.preference.PreferenceManager
import com.squareup.moshi.Moshi
import dagger.Module
import dagger.Provides
import org.greenrobot.eventbus.EventBus
import org.ligi.passandroid.json_adapter.ColorAdapter
import org.ligi.passandroid.json_adapter.ZonedTimeAdapter
import org.ligi.passandroid.model.*
import javax.inject.Singleton

@Module
class AppModule(private val app: App) {

    @Singleton
    @Provides
    internal fun providePassStore(settings: Settings, moshi: Moshi, bus: EventBus): PassStore {
        return AndroidFileSystemPassStore(app, settings, moshi, bus)
    }

    @Singleton
    @Provides
    internal fun provideMoshi(): Moshi {
        return Moshi.Builder()
                .add(ZonedTimeAdapter())
                .add(ColorAdapter())
                .build()
    }

    @Singleton
    @Provides
    internal fun provideSettings(): Settings {
        return AndroidSettings(app)
    }

    @Singleton
    @Provides
    internal fun provideSharedPreferences(): SharedPreferences {
        return PreferenceManager.getDefaultSharedPreferences(app)
    }

    @Singleton
    @Provides
    internal fun provideBus(): EventBus {
        return EventBus.getDefault()
    }

    @Singleton
    @Provides
    internal fun provideState(): State {
        return State()
    }
}
